batch/batch/worker/worker.py

```diff
@@ -806,12 +837,13 @@ async def delete_container(self):
         if self.container_is_running():
             try:
                 log.info(f'{self} container is still running, killing crun process')
-                self.process.kill()
+                self.process.terminate()
+                self.process = None
+                await check_exec_output('crun', 'kill', '--all', self.container_name)
             except asyncio.CancelledError:
                 raise
             except Exception:
-                log.warning('while deleting container, ignoring', exc_info=True)
-        self.process = None
+                log.exception('while deleting container', exc_info=True)
```

Comment:
Should the `self.process = None` line be in a finally clause?

Reply:
I'm not sure, because if we've failed before the point of marking `self.process = None` then it really isn't `None`, and it would indicate that we've ended up with a zombie process. I would rather treat the job as still running than let go of the reference to the asyncio `Process` entirely.

Reply:
In general, I'm not sure how to handle zombie processes. This is currently doing the most aggressive thing I can think of to kill off the processes.